Core/GDCore/IDE/Project/ArbitraryFunctionsWorker.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#ifndef GDCORE_ARBITRARYFUNCTIONSWORKER_H
+#define GDCORE_ARBITRARYFUNCTIONSWORKER_H
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "GDCore/String.h"
+namespace gd {
+class EventsFunction;
+class EventsFunctionsContainer;
+class ParameterMetadata;
+}  // namespace gd
+
+namespace gd {
+
+/**
+ * \brief ArbitraryFunctionsWorker is an abstract class used to browse
+ * functions signatures and do some work on them. It can be used to implement
+ * refactoring for example.
+ *
+ * \ingroup IDE
+ */
+class GD_CORE_API ArbitraryFunctionsWorker {
```

Comment:
I would be a bit more verbose and ensure we keep the term "EventsFunctions" everywhere if we can
```suggestion
class GD_CORE_API ArbitraryEventsFunctionsWorker {
```